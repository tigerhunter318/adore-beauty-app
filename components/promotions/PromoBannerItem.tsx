import React from 'react'
import Container from '../ui/Container'
import ResponsiveImage from '../ui/ResponsiveImage'
import { getIn } from '../../utils/getIn'
import { vw } from '../../utils/dimensions'
import LoadingOverlay from '../ui/LoadingOverlay'

type PromoBannerItemProps = {
  item: any
  index: number
  onPress: (item: any) => void
  isLoading: boolean
}

const PromoBannerItem = ({ item, index, onPress, isLoading }: PromoBannerItemProps) => (
  <Container style={{ overflow: 'hidden' }} key={index} align onPress={() => onPress(item)}>
    <ResponsiveImage
      src={getIn(item, 'image') || getIn(item, 'mobile_image')}
      width={getIn(item, 'mobile_image') ? 400 : 500}
      height={350}
      useAspectRatio
      styles={{ image: { width: vw(100) - 60, resizeMode: 'contain' } }}
    />
    <LoadingOverlay active={isLoading} />
  </Container>
)

export default PromoBannerItem
