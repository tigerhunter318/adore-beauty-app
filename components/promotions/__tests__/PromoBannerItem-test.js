import React from 'react'
import renderer from 'react-test-renderer'
import PromoBannerItem from '../PromoBannerItem'

const mockData = require('./__mocks__/PromoBannerItem.json')

it(`PromoBannerItem renders content`, () => {
  const tree = renderer.create(<PromoBannerItem item={mockData} isLoading={false} />).toJSON()

  expect(tree).toMatchSnapshot()
})
