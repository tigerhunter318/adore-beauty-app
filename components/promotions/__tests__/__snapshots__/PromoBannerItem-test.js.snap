// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoBannerItem renders content 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "height": "auto",
              "width": "100%",
            },
            Object {
              "aspectRatio": 1.4285714285714286,
              "resizeMode": "contain",
              "width": 690,
            },
          ],
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        imageProps={Object {}}
        resizeMode="contain"
        source={
          Object {
            "uri": "https://images.ctfassets.net/pi28xy1s107o/3WEYZF6P8kUOXOrn3sQGVu/f2bd21256b0ed892166a9651e90e5ac2/Click_Frenzy_May_WEB_AU_Promo_FT1.jpg?fm=webp&w=1000&h=700",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
